test/e2e/beat/setup_test.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package beat
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/elastic/cloud-on-k8s/pkg/controller/beat/filebeat"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/beat"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/kibana"
+)
+
+type kbSavedObjects struct {
+	Total int `json:"total"`
+}
+
+func TestBeatKibanaRef(t *testing.T) {
+	name := "test-beat-kibanaref"
+
+	esBuilder := elasticsearch.NewBuilder(name).
+		WithESMasterDataNodes(1, elasticsearch.DefaultResources)
+
+	kbBuilder := kibana.NewBuilder(name).
+		WithNodeCount(1).
+		WithElasticsearchRef(esBuilder.Ref())
+
+	fbBuilder := beat.NewBuilder(name).
+		WithType(filebeat.Type).
```

Comment:
Should we also test other types? It would be nice to have an automation to catch potential regressions around not having any dashboards bundled. Maybe it could be the same test if we can distinguish types in the request?

Reply:
I added Metricbeat and Heartbeat to this test.